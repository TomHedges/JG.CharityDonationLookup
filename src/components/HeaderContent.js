import React, { Component } from "react";
import { hot } from "react-hot-loader";
import PropTypes from "prop-types";
import "../styles/jgdr_styles.css";
import jg_logo from "./../images/jg_logo.svg";

const HeaderContent = props => (
  <div className="header_content side_pad">
    <img src={jg_logo} className="jg_logo" />
    <h1>{props.heading}</h1>
  </div>
);

HeaderContent.propTypes = {
  heading: PropTypes.string.isRequired
};

export default hot(module)(HeaderContent);
