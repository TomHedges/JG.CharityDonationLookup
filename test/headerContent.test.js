import HeaderContent from "../src/components/HeaderContent";

/**
 * Initialize common properties to be passed
 * @param {*} props properies to be override
 */
function createTestProps(props) {
  return {
    heading: "JG Donations Report",
    ...props
  };
}

let wrapper, props;

beforeEach(() => {
  props = createTestProps();
  wrapper = shallow(<HeaderContent {...props} />);
});

describe("<HeaderContent /> rendering", () => {
  it("Should render <HeaderContent /> component", () => {
    expect(wrapper.exists()).toBe(true);
  });
  it("Should render one <img>", () => {
    expect(wrapper.find("img")).toHaveLength(1);
  });
  it("Should render one <h1>", () => {
    expect(wrapper.find("h1")).toHaveLength(1);
  });
});

describe("Snapshot should contain logo and heading", () => {
  const tree = renderer.create(wrapper).toJSON();
  expect(tree).toMatchSnapshot();
});
