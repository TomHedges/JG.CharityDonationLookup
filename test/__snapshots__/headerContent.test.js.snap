// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderContent with heading and logo Should should display heading and logo 1`] = `
<div
  className="header_content side_pad"
>
  <img
    className="jg_logo"
    src={[Function]}
  />
  <h1>
    JG Donations Report
  </h1>
</div>
`;
